server/src/processing/TableOfContentsDetectionModule/TableOfContentsDetectionModule.ts

```diff
@@ -34,11 +35,19 @@ export class TableOfContentsDetectionModule extends Module<Options> {
   }
 
   public main(doc: Document): Document {
```

Comment:
@dafelix42  

135 lines in main function is not best way to code....

Could you please try to split main into small functions in order to make code more human understandable??